tests/suites/test_suite_pk.function

```diff
@@ -736,10 +812,14 @@ void pk_rsa_encrypt_test_vec( data_t * message, int mod, int radix_N,
                             output, &olen, sizeof( output ),
                             mbedtls_test_rnd_pseudo_rand, &rnd_info ) == ret );
     TEST_ASSERT( olen == result->len );
+#if !defined(MBEDTLS_USE_PSA_CRYPTO)
```

Comment:
Without this check this test case has no point. Maybe it would be better to disable it when `PSA_CRYPTO` is used?
Then for `PSA_CRYPTO` we can have known-answer test for decryption, and an encrypt-decrypt cycle for encryption.

Alternatively there is also an option to control rng with PSA like here: 
https://github.com/ARMmbed/mbedtls/blob/6d2479516c8373ed8a335f664ee4e05c4d12dd62/tests/suites/test_suite_psa_crypto_init.function#L263-L278

But I think we need input from @mpg about the testing approach.

Reply:
@mprse I was asking myself the same question, I think disable would be the best, but loosing tests coverage is not something wanted.

I already introduced a new encrypt-decrypt test in 56e27df, and decryption tests all work as-is with PSA.

I'll wait for @mpg input here.

Reply:
Thanks for introducing the encrypt-decrypt test, that's valuable to have either way. I'm a bit conflicted on this, because on one hand we obviously want `USE_PSA` to be at least well tested as the existing, OTOH, I'm not sure I'm a big fan of the principle of the existing test (and yes I know, I wrote it, but that was a long time ago, I like to think I've learned a few things in the meantime).

My main beef with the existing test is that it relies on assumptions about how the RNG is used internally. In principle in a future version could start using the RNG in other ways for other purposes such as blinding. This seems very unlikely considering this is a public key operation, but still, as a matter of principle I'm no longer that comfortable with this test.

Also, unless I'm mistaken, if we used a fixed RNG seed for PSA, we'd sill have issues such as "is CTR-DRBG or HMAC-DRBG used in this build?" and so on, so we'd end up with maybe 3 values for the "known answer" depending on the configuration, which seems more effort than warranted by the pretty low value that this test adds compared to just having an encrypt-decrypt test (plus the existing non-problematic know-answer decrypt test).

So, right now I'm inclined to ditch this test altogether, replace with the encrypt-decrypt test added by this PR.